.gitignore

```diff
@@ -0,0 +1,3 @@
+target/
+Cargo.lock
```

Comment:
I think everything here will be binary crates, so we probably want Cargo.lock under source control, no?

Reply:
> I think everything here will be binary crates, so we probably want Cargo.lock under source control, no?

Makes sense, yes. Will try to move this as well.

Reply:
I see there is a `main.rs` file. Are we sure we do not want the Cargo.lock?

Reply:
> I see there is a `main.rs` file. Are we sure we do not want the Cargo.lock?

I way I understood it, Cargo.lock file is an automatically updated file by cargo and have all details about dependencies. Since each vhost-device is going to live in its own repo, like src/i2c/, I thought it would make sense to move Cargo.lock files to src/i2c/Cargo.lock locations. Is that understanding incorrect?

Reply:
Yes, that makes sense. I misunderstood the previous comment. I thought you want to remove `Cargo.lock` :laughing:  sorry about that.

Reply:
> I think everything here will be binary crates, so we probably want Cargo.lock under source control, no?

Okay, I made a mistake somehow and though Cargo.lock is part of the initial commit :(

Cargo.lock is generated during build and all above change is doing is adding it to .gitignore so "git status" doesn't complain about it.

What else should we do here ?

Reply:
> The way I understood it, Cargo.lock file is an automatically updated file by cargo and have all details about dependencies. Since each vhost-device is going to live in its own repo, like src/i2c/, I thought it would make sense to move Cargo.lock files to src/i2c/Cargo.lock locations. Is that understanding incorrect?

It looks like Cargo [uses one `Cargo.lock` for every crate in a workspace](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html).

> Cargo.lock is generated during build and all above change is doing is adding it to .gitignore so "git status" doesn't complain about it.

We definitely want to include a Cargo.lock somewhere, as it ensures that dependencies don't get updated unexpectedly, causing unexpected breakage. (Note that this is different between libraries and binaries; for libraries, we need to support a wide range of versions, and so generally we don't check in the Cargo.lock. But for binaries, we do.)

Reply:
> It looks like Cargo [uses one `Cargo.lock` for every crate in a workspace](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html).

Yes, there is only one Cargo.lock for the entire repository.

> We definitely want to include a Cargo.lock somewhere

It is still getting generated, I am not removing it.

> as it ensures that dependencies don't get updated unexpectedly, causing unexpected breakage. (Note that this is different between libraries and binaries; for libraries, we need to support a wide range of versions, and so generally we don't check in the Cargo.lock. But for binaries, we do.)

Right, so this is still happening.